### ISSUE TITLE
Scatterplots from dataframes with multiindex columns fail

### ISSUE THREAD

**Issue by macraild** [mwaskom/seaborn#2650]
Given a long-form dataframe with multiindex columns, categorical plots accept tuple column indexers as x and y as expected, but scatterplots fail:

```
>>> sns.__version__
'0.11.0'
>>> a = np.random.rand(100)
>>> b = np.random.rand(100) + 1
>>> c = ['foo']*50 + ['bar']*50
>>> df = pd.DataFrame(data=[a,b,c], index=pd.MultiIndex.from_tuples([('1','A'), ('1', 'B'), ('2', 'C')]))
>>> # transpose so we have multiindex columns:
>>> df = df.T
>>> df = df.convert_dtypes()
>>> sns.boxplot(x=('2', 'C'), y=('1', 'A'), data=dict(df), order=['foo', 'bar'])
```
works as expected, but
```
>>> sns.scatterplot(x=('1', 'A'), y=('1', 'B'), data=df)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\macraild\Anaconda3\envs\ideomDB\lib\site-packages\seaborn\_decorators.py", line 46, in inner_f
    return f(**kwargs)
  File "C:\Users\macraild\Anaconda3\envs\ideomDB\lib\site-packages\seaborn\relational.py", line 794, in scatterplot
    p = _ScatterPlotter(
  File "C:\Users\macraild\Anaconda3\envs\ideomDB\lib\site-packages\seaborn\relational.py", line 580, in __init__
    super().__init__(data=data, variables=variables)
  File "C:\Users\macraild\Anaconda3\envs\ideomDB\lib\site-packages\seaborn\_core.py", line 604, in __init__
    self.assign_variables(data, variables)
  File "C:\Users\macraild\Anaconda3\envs\ideomDB\lib\site-packages\seaborn\_core.py", line 667, in assign_variables
    plot_data, variables = self._assign_variables_longform(
  File "C:\Users\macraild\Anaconda3\envs\ideomDB\lib\site-packages\seaborn\_core.py", line 902, in _assign_variables_longform
    raise ValueError(err)
ValueError: Length of tuple vectors must match length of `data` when both are used, but `data` has length 100 and the vector passed to `x` has length 2.
```

**Comment by mwaskom** [comment]
You are using an old version of seaborn.

**Comment by macraild** [comment]
Sorry, I didn't realise I was working from an outdated env. Fixed in 0.11.2. Closing
